components/wso2is.key.manager.operations.endpoint/src/main/java/org/wso2/is/key/manager/operations/endpoint/dcr/service/DCRMService.java

```diff
@@ -910,9 +910,23 @@ public ExtendedApplication updateApplicationOwner(String applicationOwner, Strin
         // Update Service Provider
         ServiceProvider sp = getServiceProvider(appDTO.getApplicationName(), tenantDomain);
         sp.setOwner(User.getUserFromUserName(applicationOwner));
+
+        String previousOwner = MultitenantUtils.getTenantAwareUsername(appDTO.getUsername());
         updateServiceProvider(sp, tenantDomain, MultitenantUtils.getTenantAwareUsername(appDTO.getUsername()));
         appDTO.setUsername(applicationOwner);
 
+        String newApplicationName = "";
+        if (!previousOwner.equals(applicationOwner)) {
```

Comment:
Could this lead to a NPE ? If so better to use StringUtils.equals

Reply:
This is done by https://github.com/wso2-extensions/apim-km-wso2is/pull/72